networkit/_NetworKit.pyx

```diff
@@ -3134,6 +3134,47 @@ cdef class GraphReader:
 			result = move(self._this.read(cpath)) # extra move in order to avoid copying the internal variable that is used by Cython
 		return Graph(0).setThis(result)
 
+cdef extern from "<networkit/io/GraphWriter.hpp>":
+
+	cdef cppclass _GraphWriter "NetworKit::GraphWriter":
+		_GraphWriter() nogil except +
+		void write(_Graph G, string path) nogil except +
+
+cdef class GraphWriter:
+	"""
+	Abstract base class for graph writers
+	"""
+
+	cdef _GraphWriter* _this
+
+	def __init__(self, *args, **kwargs):
+		if type(self) == GraphWriter:
+			raise RuntimeError("Error, you may not use GraphReader directly, use a sub-class instead")
+
+	def __cinit__(self, *args, **kwargs):
+		self._this = NULL
+
+	def __dealloc__(self):
+		if self._this != NULL:
+			del self._this
+		self._this = NULL
+
+	def write(self, Graph G not None, path):
+		"""
+		Write the graph to a file.
+
+		Parameters
+		----------
+		G     : networkit.Graph
+			The graph to write
+		paths : str
+			The output path
+		"""
+		cdef string c_path = stdstring(path)
```

Comment:
Optional: assert if path is empty? I mean we're check that Graph's not None ...

Reply:
I agree, we should also check for that.